tests/questionTypes.js

```diff
@@ -616,41 +616,47 @@ define([
 
             util.addQuestion("Text", "question2");
             $(".add_choice").click();
+            $(".add_choice").click();
+            $(".add_choice").click();
 
             var choice = call("getCurrentlySelectedMug");
             assert(!choice.messages.get().length, "New mug should have no errors");
             assert(!choice.p.nodeID, "New mug shouldn't have an id");
             assert(!choice.p.labelItext.get(), "New mug shouldn't have a label");
-            choice.form.vellum.ensureCurrentMugIsSaved();  // force id to generate
-            clickQuestion("question1/choice1");
-
-            util.assertJSTreeState(
-                "question1",
-                "  choice1",
-                "question2"
-            );
-
-            util.deleteQuestion("question1/choice1");
-
-            $(changerSelector + " > a").click();
-            $options = $(changerSelector + " .change-question");
-            $options.filter("[data-qtype='Text']").click();
-            assert.strictEqual($(".add_choice").length, 0);
+            _.defer(function() {
```

Comment:
Since this is a synchronous test, it looks like the assertions inside this `defer` block are likely called outside the scope of the test, which means that failure will either be silently ignored (hopefully not), or mocha may holler about failing assertions outside a test. The easy way to test this is to add a failing assertion inside the `defer` block and then run the test to see what happens. The solution is likely to add a `done` argument to the test function (line 606), making it an async test, and then to call `done();` as the last statement in this `defer`'d function.